src/features/auth/components/AuthForm/AuthForm.tsx

```diff
@@ -0,0 +1,110 @@
+'use client';
+
+import './AuthForm.scss';
+
+import { Routes } from '@core/values';
+import { Credentials } from '@entities/session';
+import { AuthFormConfig, AuthFormType, RegexpValidation, ValidationMessages } from '@features/auth';
+import { useLoader } from '@features/loader';
+import { yupResolver } from '@hookform/resolvers/yup';
+import { TextField } from '@shared/ui/TextField';
+import clsx from 'clsx';
+import Link from 'next/link';
+import { FC, memo } from 'react';
+import { useForm } from 'react-hook-form';
+import { object, string } from 'yup';
+
+type Props = {
+  type: AuthFormType;
+  onFormSubmit: (data: Credentials, reset: () => void) => void;
+};
+
+const AuthForm: FC<Props> = function ({ type, onFormSubmit }) {
+  const { isLoading } = useLoader();
+
+  const config = AuthFormConfig[type];
+
+  const emailSchema = string()
+    .required(ValidationMessages.required)
+    .matches(new RegExp(RegexpValidation.Email), ValidationMessages.email);
+
+  const usernameSchema = string()
+    .required(ValidationMessages.required)
+    .min(3, ValidationMessages.min);
+
+  const passwordSchema = string()
+    .required(ValidationMessages.required)
+    .min(8, ValidationMessages.min)
+    .matches(new RegExp(RegexpValidation.NumberContains), ValidationMessages.numberContains)
+    .matches(new RegExp(RegexpValidation.UppercaseLetterContains), ValidationMessages.upperCase)
+    .matches(new RegExp(RegexpValidation.LowercaseLetterContains), ValidationMessages.lowerCase);
+
+  const schema = object({
+    email: config.emailControl ? emailSchema : string(),
+    username: config.usernameControl ? usernameSchema : string(),
+    password: config.passwordControl ? passwordSchema : string(),
+  }).required();
```

Comment:
The general scheme will have to be left, since it depends on the props, but I think it can be taken out to useEffect to create once when mounting the component

Reply:
![image](https://user-images.githubusercontent.com/107808327/221673699-96855578-ca44-4ea4-b68b-64e95b123cc5.png)